packages/base/test/map.test.js

```diff
@@ -25,6 +25,40 @@ describe('map', () => {
     deepStrictEqual(output, ['foo_a', 'foo_b'])
   })
 
+  it('should pass optional arguments to mapper function', async () => {
+    const input = new Readable({
+      read: () => {
+        input.push('a')
+        input.push('b')
+        input.push(null)
+      },
+    })
+
+    const prependPrefix = (chunk, prefix) => prefix + chunk
```

Comment:
the additional variadic argument is not needed using [partial application](https://en.wikipedia.org/wiki/Partial_application):
```js
const prependPrefix = prefix => chunk => prefix + chunk
const outStream = input.pipe(map.call(context, prependPrefix('foo_')))
```
But expressing partial application as code snippets b59 turtle files is more cumbersome, so probably the variadic arg solution is the best option

Reply:
This is an intriguing idea but I agree that the turtle representation would likely be more complicated